Draft: Wait for WebSocket connection to open before sending

ranvit opened the pull request:
##### SUMMARY
This change enhances the WebSocket connection handling in the UI by ensuring messages are only sent when the WebSocket is fully open. This improves stability, and might help address the recurring issues of job logs failing to live-refresh

TODO: insert all the issues in operator and base repo that mention job logs not refreshing, websocket errors, DOMException errors

##### ISSUE TYPE
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
 - UI

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
v23,9.0
```


##### ADDITIONAL INFORMATION

Key Changes:
- In the onopen event, the ws.send is wrapped with a check to ensure that the WebSocket is open before sending the message. This should prevent the error related to the WebSocket being in the CONNECTING state.
- Move WebSocket creation inside connect function: This ensures a new WebSocket is instantiated during reconnection.
- Check WebSocket state before sending: Ensures messages are only sent when the WebSocket is open.
- Correctly set ws.current to a new WebSocket instance on reconnection: Ensures the reference is updated for each new connection.


<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

ranvit committed:
Wait for WebSocket before sending, create new websocket upon reconenct

ranvit committed:
Wait for WebSocket before sending, create new websocket upon reconenct

TheRealHaoLiu committed:
Fix promote from release event (#15215)

TheRealHaoLiu commented:
converting to draft PR due to title, @ranvit once its ready please put it in ready for review

hb9hnt committed:
awxkit: replace deprecated locale.format() with locale.format_string() to fix human output on Python 3.12 (#15170)

Replace deprecated locale.format with locale.format_string

This will be removed in Python 3.12 and will break human output unless fixed.

irozet12 committed:
Wrap long line to fit help window (#14597) (#15169)

Wrap long line to fit description window (#14597)

Co-authored-by: Ирина Розет <irozet@astralinux.ru>

AlexPykavy committed:
Mention in the docs that you can skip `make docker-compose-build` (#15149)

Signed-off-by: Alexander Pykavy <aleksandrpykavyj@gmail.com>

ranvit committed:
Merge branch 'ranvit-websocket-error' of https://github.com/ranvit/awx into ranvit-websocket-error
